### PR TITLE
For #15181: Remove All Downloads From Settings 

### PR DESCRIPTION
**Pull request by kglazko**
### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

File: app/src/main/java/org/mozilla/fenix/utils/Settings.kt [hunk 1]
```diff
@@ -527,6 +527,11 @@ class Settings(private val appContext: Context) : PreferencesHolder {
         default = true
     )
 
+    var deleteDownloads by booleanPreference(
+        appContext.getPreferenceKey(R.string.pref_key_delete_downloads_now),
+        default = true
+    )
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/utils/Settings.kt:533]: Let's add tests to `SettingsTest` to verify this new behaviour :)

File: app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteAndQuit.kt [hunk 2]
```diff
@@ -61,5 +62,6 @@ private suspend fun DeleteBrowsingDataController.deleteType(type: DeleteBrowsing
         DeleteBrowsingDataOnQuitType.PERMISSIONS -> withContext(IO) {
             deleteSitePermissions()
         }
+        DeleteBrowsingDataOnQuitType.DOWNLOADS -> deleteDownloads()
```
Comment (Amejia481) [inline app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteAndQuit.kt:65]: Let's add test to `DeleteAndQuitTest` to verify this new behaviour :)

**Review by Amejia481 — CHANGES_REQUESTED** [review]
It's looking good lets add some test to avoid future regressions. 👍🏽 
Just to confirm, do this functionality also have to show in the "Delete Browsing data on quit"  fragment? As I see that we are modifying the `deleteAndQuit` function but I don't it on the ui.

**Comment by Amejia481** [comment]
> It's looking good lets add some test to avoid future regressions. 👍🏽
> Just to confirm, do this functionality also have to show in the "Delete Browsing data on quit" fragment? As I see that we are modifying the `deleteAndQuit` function but I don't it on the ui.

It looks like we have the same entries: 

screenshot-1607639238028

screenshot-1607639292995

File: app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt [hunk 3]
```diff
@@ -45,6 +45,7 @@ class DeleteBrowsingDataFragment : Fragment(R.layout.fragment_delete_browsing_da
 
         controller = DefaultDeleteBrowsingDataController(
             requireComponents.useCases.tabsUseCases.removeAllTabs,
+            requireComponents.useCases.downloadUseCases.removeAllDownloads,
```
Comment (eliserichards) [inline app/src/main/java/org/mozilla/fenix/settings/deletebrowsingdata/DeleteBrowsingDataFragment.kt:48]: (This is such a small nit)
Could we make a val for `requireComponents.useCases` and `requireComponents.core` so we can just say `useCases.downloadUseCases.[...]` or `core.store`?

File: app/src/main/res/values/strings.xml [hunk 4]
```diff
@@ -1066,6 +1066,8 @@
     <string name="preferences_delete_browsing_data_cached_files_subtitle">Frees up storage space</string>
     <!-- Title for the site permissions item in Delete browsing data -->
     <string name="preferences_delete_browsing_data_site_permissions">Site permissions</string>
+    <!-- Title for the downloads item in Delete browsing data -->
```
Comment (eliserichards) [inline app/src/main/res/values/strings.xml:1069]: Could we change this description a bit? Proposal: Title for the deleting downloads item in Delete browsing data 

Update: oh strings here are all following that pattern.. we should try to change those at some point to make them a bit more descriptive 😄

**Review by eliserichards — CHANGES_REQUESTED** [review]
This looks awesome! Just two small nits and the tests Arturo mentioned, and this is good to go! 🚀

**Review by Amejia481 — APPROVED** [review]

**Review by eliserichards — APPROVED** [review]
This looks great! You did an awesome job 🎉 🎉 🎉

**Pull request merged**